### ISSUE TITLE
Pick and use an async ActiveJob adapter in production (for prints + scan processing)

### ISSUE THREAD

**Issue by mojodna** [fieldpapers/fieldpapers#3]
The app currently uses the default ActiveJob adapter (`InlineAdapter`), which runs jobs synchronously.  While this is convenient for development, it ties up a thread / process in production.

List of potential adapters: http://edgeapi.rubyonrails.org/classes/ActiveJob/QueueAdapters.html

Many of these require the use of Redis for storage of jobs, which introduces another dependency that needs to be watched.  The benefit of these (especially Sidekiq) is that one gets a status dashboard for "free."

Sucker Punch runs inline, which might be good for development (or for appliance mode) in order to implement status polling.

[shoryuken](https://github.com/phstc/shoryuken) looks like a promising one to use to avoid the typical Redis (or other) dependency, since we're assuming AWS deployment for fieldpapers.org.

By sticking with ActiveJob, installations in other environments (the appliance model, for example) can use different adapters without requiring code changes.

**Comment by mojodna** [comment]
(This is partially why the dev site shows an Application Error when one reaches the final stage of the compose wizard--it's trying to use the `InlineAdapter` (which would be fine) to run a job that calls `docker`, which isn't available on Heroku.)

**Comment by mojodna** [comment]
I added Sucker Punch via https://github.com/fieldpapers/fp-web/commit/a2eea5162fcd06cd028c8a54c85070a49966ccc8, so it will now behave asynchronously when running in a dev environment.

Because the docs on shoryuken are light on how to use it with ActiveJob (which I still think is a good idea), I'm also using sucker punch on Heroku for testing purposes.

Since PDF generation does some of its work inside a docker container, I added a Linux docker binary to `bin/linux` and some keys (BAD) that only work with a one-off docker instance launched using [docker machine](https://github.com/docker/machine).  This allows the Heroku instance to run commands inside containers (and to theoretically do other things).
